tests/unit/test_tracing.c

```diff
@@ -663,5 +665,99 @@ SENTRY_TEST(drop_unfinished_spans)
     TEST_CHECK_INT_EQUAL(called_transport, 1);
 }
 
+static void
+forward_headers_to(const char *key, const char *value, void *userdata)
+{
+    sentry_transaction_context_t *tx_ctx
+        = (sentry_transaction_context_t *)userdata;
+
+    sentry_transaction_context_update_from_header(tx_ctx, key, value);
+}
+
+SENTRY_TEST(distributed_headers)
+{
+    sentry_options_t *options = sentry_options_new();
+    sentry_options_set_dsn(options, "https://foo@sentry.invalid/42");
+
+    sentry_options_set_traces_sample_rate(options, 1.0);
+    sentry_options_set_max_spans(options, 2);
+    sentry_init(options);
+
+    sentry_transaction_context_t *tx_ctx
+        = sentry_transaction_context_new("wow!", NULL);
+    sentry_transaction_t *tx = sentry_transaction_start(tx_ctx);
+
+    const char *trace_id = sentry_value_as_string(
+        sentry_value_get_by_key(tx->inner, "trace_id"));
+    TEST_CHECK(!sentry__string_eq(trace_id, ""));
+
+    const char *span_id
+        = sentry_value_as_string(sentry_value_get_by_key(tx->inner, "span_id"));
+    TEST_CHECK(!sentry__string_eq(span_id, ""));
+
+    // check transaction
+    tx_ctx = sentry_transaction_context_new("distributed!", NULL);
+    sentry_transaction_iter_headers(tx, forward_headers_to, (void *)tx_ctx);
```

Comment:
this seems like a fairly unconventional signature given what i understand of the intent of the [api's design](https://develop.sentry.dev/sdk/performance/#new-span-and-transaction-classes): the way i read it was that you would use the headers getter/setter like so, using this test as an example:

```
headers = sentry_transaction_iter_headers(tx);
continuing_transaction = sentry_transaction_continue_from_headers(sentrytrace_header);
```

this appears to have pushed aside the idea of returning some sort of map in favour of a more "iterable"-like api. is there a particular reason why that decision was made? does it make using this more natural over returning some sort of map? 

stepping outside of the scope of the test itself, the return value is presumably going to be attached to the headers of requests (i.e. events and transactions) sent to sentry, so would this api still feel natural once we add that functionality?

Reply:
Well, maps don’t exist in C, neither do "standard" iterators.
We have a map in `sentry_value_t`, but as @flub noticed, we can’t even iterate over its keys. Also having API users go through constructing a `sentry_value_t` might be a bit inconvenient.

Reply:
I think the generic alternative would be to define a struct with key, value, next and force people to populate that.  but really it's no better than this way i guess, passing the userdata around saves some conversions in favour of more function calls.